packages/math/typesetter.lua

```diff
@@ -144,9 +144,15 @@ local function handleMath (_, mbox, mode)
   if mode == "display" then
     SILE.typesetter:endline()
     SILE.typesetter:pushExplicitVglue(SILE.settings:get("math.displayskip"))
-    SILE.call("center", {}, function()
-      SILE.typesetter:pushHorizontal(mbox)
-    end)
+    SILE.typesetter:pushGlue(SILE.nodefactory.hfillglue())
+    SILE.typesetter:pushHorizontal(mbox)
+    SILE.typesetter:pushGlue(SILE.nodefactory.hfillglue())
+    if counter then
+      SILE.call("increment-counter", { id=counter } )
+      SILE.call("math-counterstyle", { id=counter } )
+    else
+      SILE.call("hbox")
```

Comment:
I suppose you did that to cancel the fact that regular glues are ignore at an end of line?
If so, i think this empty hbox trick can be avoid, if the glue is made explicit (i.e. on line 149 above, use `pushExplicitGlue`)

Reply:
Checking how LaTeX behaves in justified/raggedleft/raggedright/center - additionally nested in a block (quote)[^1] 

![image](https://github.com/sile-typesetter/sile/assets/18075640/bd2e79cc-c019-4092-888c-eabb18ee2b25)

[^1]: problematic with our current implementation of ragged environments (not nesting, not keeping parindent, so basically behaving differently, in a questionable way)

We should design a test scenario and check how we compare...

Reply:
Thanks for the `pushExplicitGlue` suggestion -- I didn't knew that!

As a (really) side note, since you brought the ragged environments topic: when you suggested to use options to number equations in the display environment, I also wondered: what if we had an option to place the number on the left, or on the right side? Playing a bit, I saw that, surprisingly (to me), documents with `direction=RTL` already place the numbering on the left side (although the equation may be bugged -- eq. 1 is not centered, and parenthesis in equation 2 are wrong)

![math-RTL](https://github.com/sile-typesetter/sile/assets/3310604/e44f1ed2-78f4-4f76-a5d4-6444c313b45c)

Since left numbered equations are not my use case, I opted to not add any option in this regard.

Reply:
So let's craft a regression test similar to my above LaTeX tests. That test eventually should be committed in `tests/math-display-numbered.sil` and has it's `-b debug` result also committed as `tests/math-display-numbered.expected`

[math-display-numbered.zip](https://github.com/sile-typesetter/sile/files/12837360/math-display-numbered.zip)

Before any change:

![image](https://github.com/sile-typesetter/sile/assets/18075640/e57abf12-ef6d-444a-8cbb-85833ced763f)

After your changes as of c4db864 

![image](https://github.com/sile-typesetter/sile/assets/18075640/465b958f-ac15-4ee1-9fdd-a42c129a55f7)

**It breaks expectations with both numbered and unnumbered display math.** That tells the importance of having (non-)regressing tests for features. :smile_cat: 

What I would expect, _at least_, is:

![image](https://github.com/sile-typesetter/sile/assets/18075640/3d78ea4d-f82f-4bac-b340-60302728ffba)

Which I achieved hacking your proposal as follows:
(N.B. it could possibly be simplified, it basically re-implements or rather adapts a raggedright environment, respecting possible nesting)[^1]

```
  if mode == "display" then
    SILE.typesetter:endline()
    SILE.typesetter:pushExplicitVglue(SILE.settings:get("math.displayskip"))
    SILE.settings:temporarily(function ()
      local lskip = SILE.settings:get("document.lskip") or SILE.nodefactory.glue()
      local rskip = SILE.settings:get("document.rskip") or SILE.nodefactory.glue()
      SILE.settings:set("document.lskip", SILE.nodefactory.hfillglue(lskip.width.length))
      SILE.settings:set("document.rskip", SILE.nodefactory.glue(rskip.width.length))
      SILE.settings:set("document.parindent", SILE.nodefactory.glue())
      SILE.settings:set("current.parindent", SILE.nodefactory.glue())
      SILE.settings:set("typesetter.parfillskip", SILE.nodefactory.glue())
      SILE.settings:set("document.spaceskip", SILE.length("1spc"))

      SILE.typesetter:pushHorizontal(mbox)
      SILE.typesetter:pushExplicitGlue(SILE.nodefactory.hfillglue())

      if counter then
        SILE.call("increment-counter", { id=counter } )
        SILE.call("math:numberingstyle", { id=counter } )
      end
      SILE.typesetter:endline()
    end)
    SILE.typesetter:pushExplicitVglue(SILE.settings:get("math.displayskip"))
 ```

_But the devil is in the details_, and as you noted it probably doesn't work well in RTL...

Also, what happens if the equation is long and takes up most of the line so the fill and the number cannot fit? This would also call from a dedicated test.

[^1]: The current implementation of `\raggedright` in SILE 0.14 is subpar, and I might suggest at some point to go for the re-implementation I made in my silex.sile module. Whatever, here for the purpose of display math, it's probably safer _anyway_ not to depend from it.